dataset/histogram.cpp

```diff
@@ -117,7 +117,7 @@ DataArray histogram(const DataArrayConstView &events,
                        transform_flags::expect_no_variance_arg<3>});
       },
       dim_of_coord(events.coords()[dim], dim), dim, binEdges);
-  result.setCoord(dim, binEdges);
+  result.coords().set(dim, binEdges);
```

Comment:
`setCoord` (and `setAttr` and `setMasks`) only existings for `Dataset` and is needed there since it is actually called by `coords().set()`. `coords().set()` is the "newer" way and also works for `DataArray`. You are right that it may be a good idea to standardize on the latter. I am adding a new item to the task list #1047.